typescript/cli/src/tests/warp/warp-rebalancer.e2e-test.ts

```diff
@@ -182,26 +182,31 @@ describe('hyperlane warp rebalancer e2e tests', async function () {
       REBALANCER_STRATEGY_CONFIG_PATH,
     );
 
-    return new Promise(async (resolve, reject) => {
+    return new Promise((resolve, reject) => {
+      const dataHandler = async (chunk: any) => {
+        if (chunk.includes(log)) {
+          clearTimeout(timeoutId);
+          process.stdout.off('data', dataHandler);
+          resolve();
+          void process.kill();
+        }
+      };
+
       const timeoutId = setTimeout(async () => {
-        await process.kill();
+        process.stdout.off('data', dataHandler);
         reject(new Error(`Timeout waiting for log: "${log}"`));
+        void process.kill();
       }, timeout);
 
       process.catch((e) => {
         clearTimeout(timeoutId);
-        // TODO: Do a pretty print of the error
-        reject(e.text());
+        process.stdout.off('data', dataHandler);
+        reject(
+          new Error(`Process failed before logging: "${log}" with error: ${e}`),
+        );
       });
 
-      for await (const chunk of process.stdout) {
-        if (chunk.includes(log)) {
-          clearTimeout(timeoutId);
-          resolve();
-          await process.kill();
-          break;
-        }
-      }
+      process.stdout.on('data', dataHandler);
```

Comment:
I went this exact same path, but I rather keep what's currently done with the `for await` for `chunks`.

Reply:
But the for await requires the promise executor to be async and the linter doesn't like that